### ISSUE TITLE
[2.7] test_release_task_refs() of test_multiprocessing.py is unstable

### ISSUE THREAD

**Issue by vstinner** [python/cpython#74455]
BPO | [30269](https://bugs.python.org/issue30269)
--- | :---
Nosy | @vstinner
Superseder | <li>bpo-29861: multiprocessing Pool keeps objects (tasks, args, results) alive too long</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-05.07:40:55.029>
created_at = <Date 2017-05-04.17:19:01.473>
labels = ['tests']
title = '[2.7] test_release_task_refs() of test_multiprocessing.py is unstable'
updated_at = <Date 2017-05-05.07:40:55.027>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-05.07:40:55.027>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-05-05.07:40:55.029>
closer = 'vstinner'
components = ['Tests']
creation = <Date 2017-05-04.17:19:01.473>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30269
keywords = []
message_count = 2.0
messages = ['293003', '293060']
nosy_count = 1.0
nosy_names = ['vstinner']
pr_nums = []
priority = 'normal'
resolution = 'duplicate'
stage = 'resolved'
status = 'closed'
superseder = '29861'
type = None
url = 'https://bugs.python.org/issue30269'
versions = ['Python 2.7']
```

</p></details>

**Comment by vstinner** [comment]
[2.7] test_release_task_refs() of test_multiprocessing.py fails randomly.

Example:

http://buildbot.python.org/all/builders/x86%20Ubuntu%20Shared%202.7/builds/136/steps/test/logs/stdio

```pytb
[116/401/1] test_multiprocessing
[77069 refs]
test test_multiprocessing failed -- Traceback (most recent call last):
  File "/srv/buildbot/buildarea/2.7.bolen-ubuntu/build/Lib/test/test_multiprocessing.py", line 1293, in test_release_task_refs
    self.assertEqual(set(wr() for wr in refs), {None})
AssertionError: Items in the first set but not the second:
<test.test_multiprocessing.CountedObject object at 0x40f7061c>
<test.test_multiprocessing.CountedObject object at 0x40f70654>
<test.test_multiprocessing.CountedObject object at 0x40f70a7c>
<test.test_multiprocessing.CountedObject object at 0x40f70cac>
<test.test_multiprocessing.CountedObject object at 0x40f709d4>
<test.test_multiprocessing.CountedObject object at 0x40f70734>
<test.test_multiprocessing.CountedObject object at 0x40f70d54>
<test.test_multiprocessing.CountedObject object at 0x40f707a4>
<test.test_multiprocessing.CountedObject object at 0x40f70ea4>
<test.test_multiprocessing.CountedObject object at 0x40f705e4>
Items in the second set but not the first:
None
(...)
Re-running test 'test_multiprocessing' in verbose mode
(...)
Ran 167 tests in 20.640s
OK (skipped=12)
```

**Comment by vstinner** [comment]
Oh, it's just a missed backport of the last bpo-29861 fix, only 2.7 miss a backport.
